cargo-dist/src/backend/ci/github.rs

```diff
@@ -164,13 +175,33 @@ impl GithubCiInfo {
         let pr_run_mode = dist.pr_run_mode;
 
         let tap = dist.tap.clone();
-        let plan_jobs = dist.plan_jobs.clone();
-        let local_artifacts_jobs = dist.local_artifacts_jobs.clone();
-        let global_artifacts_jobs = dist.global_artifacts_jobs.clone();
-        let host_jobs = dist.host_jobs.clone();
+
+        let mut job_permissions = dist.github_custom_job_permissions.clone();
+        // user publish jobs default to elevated priviledges
+        for name in &dist.user_publish_jobs {
+            job_permissions.entry(name.clone()).or_insert_with(|| {
+                GithubPermissionMap::from_iter([
+                    ("id-token".to_owned(), GithubPermission::Write),
+                    ("packages".to_owned(), GithubPermission::Write),
+                ])
+            });
+        }
+
+        let mut root_permissions = GithubPermissionMap::new();
+        root_permissions.insert("contents".to_owned(), GithubPermission::Write);
+        if github_attestations {
+            root_permissions.insert("attestations".to_owned(), GithubPermission::Write);
```

Comment:
Attestations use two permissions, not one: it also needs `id-token`. https://github.com/axodotdev/cargo-dist/blob/4b8b18d315a92d1c4aac3063ff01acd0a5e4b250/cargo-dist/templates/ci/github/release.yml.j2#L30-L31